src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -13,9 +13,11 @@
 #define UCT_DC_MLX5_IFACE_TXQP_GET(_iface, _ep, _txqp, _txwq) \
 { \
     uint8_t dci; \
-    dci = (_ep)->dci; \
-    _txqp = &(_iface)->tx.dcis[dci].txqp; \
-    _txwq = &(_iface)->tx.dcis[dci].txwq; \
+    dci                     = (_ep)->dci; \
+    _txqp                   = &(_iface)->tx.dcis[dci].txqp; \
+    _txwq                   = &(_iface)->tx.dcis[dci].txwq; \
+    (_txwq)->fi.fence_flag |= ((_ep)->flags & UCT_DC_MLX5_EP_FLAG_FENCE) ? 1 : 0; \
```

Comment:
maybe use `=` and `!!` instead of  `|=` and `?`

Reply:
nop, fence_flag should not be reset if it is enabled before

Reply:
maybe `!!((_txwq)->fi.fence_flag || ((_ep)->flags & UCT_DC_MLX5_EP_FLAG_FENCE))`